packages/@aws-cdk/aws-codedeploy/lib/deployment-configuration.ts

```diff
@@ -0,0 +1,155 @@
+        implements ServerDeploymentConfiguration {
+    /**
+     * Import a custom Deployment Configuration for an EC2/on-premise Deployment Group defined either outside the CDK,
+     * or in a different CDK Stack and exported using the {@link #export} method.
+     *
+     * @param parent the parent Construct for this new Construct
+     * @param id the logical ID of this new Construct
+     * @param props the properties of the referenced custom Deployment Configuration
+     * @returns a Construct representing a reference to an existing custom Deployment Configuration
+     */
+    public static import(parent: cdk.Construct, id: string, props: CustomServerDeploymentConfigurationRefProps):
+            CustomServerDeploymentConfigurationRef {
+        return new ImportedCustomServerDeploymentConfigurationRef(parent, id, props);
+    }
+
+    public abstract readonly deploymentConfigurationName: DeploymentConfigurationName;
+    public abstract readonly deploymentConfigurationArn: DeploymentConfigurationArn;
+
+    public export(): CustomServerDeploymentConfigurationRefProps {
+        return {
+            deploymentConfigurationName: new cdk.Output(this, 'DeploymentConfigurationName', {
+                value: this.deploymentConfigurationName,
+            }).makeImportValue(),
+        };
+    }
+}
+
+class ImportedCustomServerDeploymentConfigurationRef extends CustomServerDeploymentConfigurationRef {
+    public readonly deploymentConfigurationName: DeploymentConfigurationName;
+    public readonly deploymentConfigurationArn: DeploymentConfigurationArn;
+
+    constructor(parent: cdk.Construct, id: string, props: CustomServerDeploymentConfigurationRefProps) {
+        super(parent, id);
+
+        this.deploymentConfigurationName = props.deploymentConfigurationName;
+        this.deploymentConfigurationArn = deploymentConfigurationName2Arn(this.deploymentConfigurationName);
+    }
+}
+
+/**
+ * The possible representations of a healthy-host value -
+ * either percent-based, or an absolute number.
+ */
+export enum HostsRepresentation {
+    NUMBER = 'HOST_COUNT',
+    PERCENTAGE = 'FLEET_PERCENT'
+}
+
+export interface MinimumHealthyHosts {
+    readonly representation: HostsRepresentation;
+
+    readonly value: number;
+}
+
+/**
+ * Construction properties of {@link CustomServerDeploymentConfiguration}.
+ */
+export interface CustomServerDeploymentConfigurationProps {
+    /**
+     * The physical, human-readable name of the Deployment Configuration.
+     *
+     * @default a name will be auto-generated
+     */
+    deploymentConfigurationName?: string;
+
+    /**
+     * The minimum healhty hosts setting for this EC2/on-premise Deployment Configuration.
+     */
+    minimumHealthyHosts: MinimumHealthyHosts;
+}
+
+/**
+ * A custom Deployment Configuration for an EC2/on-premise Deployment Group.
+ */
+export class CustomServerDeploymentConfiguration extends CustomServerDeploymentConfigurationRef {
```

Comment:
Should should be called `DeploymentConfig` and `DeploymentConfigRef` (to align with the CFN resource name).

Reply:
Fair enough. I wanted to make it more consistent than CFN, because, for instance, the auto-rollback property and type is called `AutoRollbackConfiguration`, and it seems weird to me to have `Configuration` and `Config` both used in the same class. But I can rename it if you like `Config` here better.

Reply:
Oh, also, the `Server` prefix is required (other compute platform like Lambda have completely different Deployment Configurations - it should not be possible to mix them).